source/common/upstream/health_checker_impl.h

```diff
@@ -252,6 +252,9 @@ class TcpHealthCheckerImpl : public HealthCheckerImplBase {
     TcpHealthCheckerImpl& parent_;
     Network::ClientConnectionPtr client_;
     std::shared_ptr<TcpSessionCallbacks> session_callbacks_;
+    // If true, stream reset was initiated by us, not e.g. remote reset.
+    // In this case healthcheck status already reported, only state cleanup required.
+    bool expect_reset_{};
```

Comment:
nit: can you make this `expect_close_` and update the comment above to talk about close vs. reset? There is no reset happening for TCP.

Reply:
Done